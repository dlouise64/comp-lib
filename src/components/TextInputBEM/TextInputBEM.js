import React from 'react';
import PropTypes from 'prop-types';
import Label from '../Label';

/** Text input */
function TextInputBEM({htmlId, name, label, type="text", required=false, onChange, placeholder, value, error, children, ...props}) {
  return (
    <div className="textInput">
      <Label htmlFor={htmlId} label={label} required={required} />
      <input
        id={htmlId}
        type={type}
        name={name}
        placeholder={placeholder}
        value={value}
        onChange={onChange}
        className={error && 'textInput__input--state-error'}
        {...props} />
        {children}
        { error && <div className="textInput__error" style={{color: 'red'}}>{error}</div>}
    </div>
  )
}

TextInputBEM.propTypes = {
  /** Unique HTML ID. Used for tying label to HTML input. */
  htmlId: PropTypes.string.isRequired,

  /** Input name */
  name: PropTypes.string.isRequired,

  /** Input Label */
  label: PropTypes.string.isRequired,

  /** Input type */
  type: PropTypes.oneOf(['text', 'number', 'password']),

  /** Mark label with asterisk if set to true */
  required: PropTypes.bool,

  /** function to call onChange */
  onChange: PropTypes.func.isRequired,

  /** Placeholder to display when empty */
  placeholder: PropTypes.string,

  /** value */
  value: PropTypes.any,

  /** String to display when error occurs */
  error: PropTypes.string,

  /** Child component to display next to input */
  children: PropTypes.node
};

export default TextInputBEM;
