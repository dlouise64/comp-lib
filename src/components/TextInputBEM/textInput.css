.textInput {
  margin-bottom: 16px;
}

.textInput__error {
  color:red;
}

.textInput__input--state-error {
  border: 1px solid red;
}
