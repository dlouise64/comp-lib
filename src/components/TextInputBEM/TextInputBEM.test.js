import React from 'react';
import renderer from 'react-test-renderer';
import TextInputBEM from './TextInputBEM';

test('<TextInputBEM /> renders correctly', () => {
  const tree = renderer.create(<TextInputBEM
    htmlId="my_id"
    name="my_name"
    label="the label"
    onChange={() => {}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
