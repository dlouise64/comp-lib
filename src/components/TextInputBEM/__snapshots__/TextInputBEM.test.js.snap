// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInputBEM /> renders correctly 1`] = `
<div
  className="textInput"
>
  <label
    htmlFor="my_id"
    style={
      Object {
        "display": "block",
      }
    }
  >
    the label
     
  </label>
  <input
    id="my_id"
    name="my_name"
    onChange={[Function]}
    type="text"
  />
</div>
`;
